default/python/AI/test/test_universe.py

```diff
@@ -503,6 +505,278 @@ def setUp(self):
             self.objects_to_test.extend([fo.getPartType(part) for part in design.parts if part])
 
 
+class HullTypeTester(PropertyTester):
+    class_to_test = fo.hullType
+    properties = deepcopy(PropertyTester.properties)
+    properties.update({
+        "name": {
+            TYPE: str,
+        },
+        "numSlots": {
+            TYPE: int,
+        },
+        "structure": {
+            TYPE: float,
+        },
+        "stealth": {
+            TYPE: float,
+        },
+        "fuel": {
+            TYPE: float,
+        },
+        "speed": {
+            TYPE: float,
+        },
+        "slots": {
+            TYPE: fo.ShipSlotVec,
+        },
+        "costTimeLocationInvariant": {
+            TYPE: bool,
+        },
+    })
+
+    def test_numSlotsOfSlotType(self):
+        for obj in self.objects_to_test:
+            total_slots = 0
+            for slot_type in (fo.shipSlotType.internal, fo.shipSlotType.external, fo.shipSlotType.core):
+                retval = obj.numSlotsOfSlotType(slot_type)
+                self.assertIsInstance(retval, int)
+                self.assertGreaterEqual(retval, 0)
+                total_slots += retval
+            self.assertEquals(total_slots, obj.numSlots)
+
+            with self.assertRaises(Exception):
+                obj.numSlotsOfSlotType(2)
+            with self.assertRaises(Exception):
+                obj.numSlotsOfSlotType("")
+
+    def test_productionCost(self):
+        for obj in self.objects_to_test:
+            retval = obj.productionCost(fo.empireID(), INVALID_ID)
```

Comment:
I'll probably merge some changes soon that add another `INVALID_ID` parameter to this and similar calls: https://github.com/freeorion/freeorion/commit/6f797a073b3e7aada63f8b577997383007becc5e

Reply:
Thanks for the heads-up, might spare me some headache trying to figure out while some tests may suddenly fail.